src/main/java/frc/robot/subsystems/leds/LEDSubsystem.java

```diff
@@ -38,20 +50,116 @@ public void periodic() {
         // Toggle the blink boolean every duration to swap the LEDs between the driver piece color
         // and the blink color.
         if (blinkTimer.advanceIfElapsed(BLINK_DURATION_SECONDS)) blinking = !blinking;
-        ledStrip.setSolidColor(blinking ? BLINK_COLOR : color);
+        if(manual){
+            //if manual and continuous set the bottom to color
+            ledStrip.updateContinuousColor(color);
+            ledStrip.setContinuousColor();
+        //} else if(manual){
+            //if manual and not continuous fill entire strip with color
+            // ledStrip.fillContinuousColor(color);
+            // ledStrip.setContinuousColor();
+        } else {
+            setColorPulse();
+            //setTwoColor()
+        }
     }
 
     /**
-     * Sets the driver-provided color of the LEDs.
-     * @param red The red component of the color.
-     * @param green The green component of the color.
-     * @param blue The blue component of the color.
+     * Sets the driver-provided color of the LEDs from provided RGB values.
+     * @param r The red component of the color, from [0, 255].
+     * @param g The green component of the color, from [0, 255].
+     * @param b The blue component of the color, from [0, 255].
      */
-    public void setColor(double red, double green, double blue) {
+    public void setRGB(double r, double g, double b) {
+        //if the color doesnt change, dont do anything
+        if(this.color.red == r && this.color.green == g && this.color.blue == b){ 
+            return;
+        }
         this.color = new Color(
-            (int) (red * BRIGHTNESS_SCALE_FACTOR),
-            (int) (green * BRIGHTNESS_SCALE_FACTOR),
-            (int) (blue * BRIGHTNESS_SCALE_FACTOR)
+            (int) (r * BRIGHTNESS_SCALE_FACTOR),
+            (int) (g * BRIGHTNESS_SCALE_FACTOR),
+            (int) (b * BRIGHTNESS_SCALE_FACTOR)
+        );
+        //sets the whole strip to the color
+        ledStrip.fillContinuousColor(this.color);
```

Comment:
What Ethan said; the public methods should only modify the subsystem's state, and the LEDs should be set based on the state in `periodic()`.